src/wp-includes/nav-menu.php

```diff
@@ -580,7 +585,11 @@ function wp_update_nav_menu_item( $menu_id = 0, $menu_item_db_id = 0, $menu_item
 	if ( $update ) {
 		$post['ID']          = $menu_item_db_id;
 		$post['post_status'] = ( 'draft' === $args['menu-item-status'] ) ? 'draft' : 'publish';
-		wp_update_post( $post );
+
+		$update_post = wp_update_post( $post );
```

Comment:
For the `is_wp_error()` check to work, `wp_update_post()` should be called with the `$wp_error` parameter set to `true` (as it's `false` by default), i.e. `wp_update_post( $post, true )`.